### ISSUE TITLE
show black screen

### ISSUE THREAD

**Issue by ohouha** [GPSnoopy/RayTracingInVulkan#19]
Hi,
recently I am learning raytracing, and find your awesome git, I think your git is what I need for raytracing learning, but when I try to build it on my ubuntu system, TITAN X (Pascal), I find the screen is black , nothing rendered, and I can see the frame is around 12 fps, which I think is calculating, and I turn off the enable ray tracing check box, it's still show nothing,  I already build nvidia lessons about vulkan and raytracing example, and pretty sure it can run on my card, so is there any hint that I can check? Thank you.

-Best
Wy

**Comment by GPSnoopy** [comment]
Hi @ohouha ,

Which version are you using? 
If you have the possibility, try the Win32 prebuild binaries if you have a dual boot, to confirm that the card works fine.

Master is currently a tad broken depending on which driver you're using, as `VK_KHR_ray_tracing` specs got finalised (and well, doesn't exist in that form anymore).

I'm waiting on a proper Vulkan SDK release from LunarG to update the code to the final extension specs.

**Comment by ohouha** [comment]
Hi, 
I am using drive:450, can you recommend which version should I use?

**Comment by GPSnoopy** [comment]
I meant for the source code (should have been clearer). Try with the latest tagged release, or lookup the history of master and pick the commit before the VK_KHR_ray_tracing PR was merged in

**Comment by ohouha** [comment]
oh, I see, I will try other versions.

**Comment by GPSnoopy** [comment]
Hey @ohouha, with your current setup, you're probably best to try [Release 4](https://github.com/GPSnoopy/RayTracingInVulkan/releases/tag/r4).

I've updated master and tagged [Release 5](https://github.com/GPSnoopy/RayTracingInVulkan/releases/tag/r5) which uses the latest Vulkan ray tracing extensions, but they require NVIDIA drivers 460.89. As far as I'm aware, these newer drivers are not yet available on Linux.

**Comment by hradec** [comment]
theres a beta 460.27 released today (15-Dec-2020) together with the Win10 460.89. 
https://www.nvidia.com/download/driverResults.aspx/167671/en-us
Would that beta work? the release notes says `Add support for the VK_KHR_ray_tracing_pipeline extension.`!

**Comment by GPSnoopy** [comment]
Ah, failed to see that. Yes, that should work, in theory.

Personally, I'm lazy. I'm waiting for the official Ubuntu 20.04 packages before testing it on Linux. ;-)

**Comment by hradec** [comment]
LoL! Fair enough! 

As I'm on Arch , I'll update to the latest driver and give it a go, as soon as I finish a Dockerfile to build and run it inside docker, using ubuntu 20.04! LoL

**Comment by Skipperro** [comment]
Hi,

I've tried master branch on Linux Mint 20 with NVIDIA Titan Xp and newest drivers 460.32.03 with RayTracing extensions (https://www.nvidia.com/download/driverResults.aspx/168347/en-us) and newest Vulkan SDK.

After building the app it kind of run - my GPU is loaded, I see lower FPS values with ray-tracing enabled and higher without ray-tracing. If I use WSAD keys or move mouse FPS is also changing. But no matter what I do, all I see is black window with controls and stats.

Hope it will help you, I'm really excited to see it working :) 

![image](https://user-images.githubusercontent.com/20705647/104088067-6fc01a80-5264-11eb-9052-55163f172708.png)

**Comment by GPSnoopy** [comment]
I've just tested on Ubuntu 20.04 with an RTX 3090 and NVIDIA drivers 460.32.03, and it works as expected. I suspect this bug might only occur on Pascal GPUs. Unfortunately, my PC with the 1080 Ti is currently in storage.

@Skipperro Does it display anything if you uncheck the "Enable ray tracing" option?

Also, if you have a dual boot, could you test on Windows to see if it's only happening on Linux?

**Comment by Skipperro** [comment]
@GPSnoopy No, if I uncheck "Enable ray tracing" FPS jumps to 144 (which is my display refresh rate) but everything is still black. 

Maybe you could include Linux binaries in next release, so I can test if I didn't make any mistake while building it?

I don't have dual-boot on my PC at home, but I've just connected with one of my PCs at work (with Quadro RTX 4000) and tried Release 5 there. Got this error:
![image](https://user-images.githubusercontent.com/20705647/104108102-47ff9f80-52c2-11eb-8aa0-3bb444d537f4.png)

**Comment by GPSnoopy** [comment]
@Skipperro I'm surprised about the RTX 4000 failure. Was that through Remote Desktop or similar?

I've added prebuilt R5 binaries with debug info, such that the stack trace will display the method names (the 7zipped file is not that much bigger, so I might make this the default for future releases). This will help find out exactly where the RTX 4000 fails to enumerate a property: https://github.com/GPSnoopy/RayTracingInVulkan/releases/download/r5/RayTracingInVulkan.r5.WithDebugInfo.7z

Unfortunately without my 1080 Ti to locally debug the issue, it's going to be hard for me to reproduce it.  Best bets include:
- a motivated soul to find a fix and raise a PR,
- a future version of the Vulkan SDK with new validation errors that spot the bug,
- a future version of NVIDIA drivers (in the unlikely case it's not my fault).

The fact the the screen remains black even when not using the ray tracing pipeline implies that this is a bug in the application rather than a driver bug.

I would suggest that in the meantime you try other VK_KHR_ray_tracing_pipeline applications such as Quake II RTX and NVIDIA's own examples (https://github.com/nvpro-samples/vk_raytracing_tutorial_KHR). Just to make sure that it works correctly on your machine in general.

**Comment by GPSnoopy** [comment]
Would you mind trying Release 6 (https://github.com/GPSnoopy/RayTracingInVulkan/releases/tag/r6)?

It fixes several issues that were uncovered when testing the ray tracer on AMD RX 6000 series. One of these might be the same reason for the black screen you're seeing.
